src/__tests__/is-of-type.dts.spec.ts

```diff
@@ -0,0 +1,76 @@
+import { isOfType } from '../is-of-type'
+import { createActionCreator } from '../create-action-creator'
+
+// @dts-jest:group isOfType
+
+const increment = createActionCreator('[Counter] increment')
+const decrement = createActionCreator('[Counter] decrement')
+const reset = createActionCreator(
+  '[Counter] reset',
+  resolve => (value: number) => resolve(value)
+)
+
+{
+  // @dts-jest:pass:snap
+  isOfType(increment.type, increment())
+
+  // @dts-jest:pass:snap
+  isOfType([increment.type, reset.type], decrement())
+
+  // @dts-jest:pass:snap
+  isOfType(reset(0), reset(1))
+
+  // @dts-jest:pass:snap
+  isOfType([increment(), decrement(), reset(0)], increment())
+
+  // @dts-jest:pass:snap
+  isOfType(decrement, increment())
+
+  // @dts-jest:pass:snap
+  isOfType([decrement, reset], reset(0))
+
+  // @dts-jest:pass:snap
+  isOfType([increment.type, decrement(), reset], increment())
+
+  // @dts-jest:pass:snap
+  isOfType([increment.type, reset], decrement())
```

Comment:
These tests look useless. The important job of `isOfType` function is to change the input action type to possible cases. So the dts test should check the input action type after `isOfType` function application.

Reply:
Is this understanding of mine correct?

```ts
const increment = createActionCreator('[Counter] increment')
const decrement = createActionCreator('[Counter] decrement')
const reset = createActionCreator(
  '[Counter] reset',
  resolve => (value: number) => resolve(value)
)
const actions = [
  increment,
  decrement,
  reset
]

const actual = actions.filter(
  isOfType(increment)
);
// @dts-jest:pass:snap
actual
```

Reply:
This looks awesome. Let's do this.

Also, there is no need for `actual` const.
 
```ts
// @dts-jest:pass:snap
actions.filter(isOfType(increment.type))
```

Reply:
completed!